src/fr/scanmanga/src/eu/kanade/tachiyomi/extension/fr/scanmanga/ScanManga.kt

```diff
@@ -1,217 +1,251 @@
 package eu.kanade.tachiyomi.extension.fr.scanmanga
 
+import android.util.Base64
 import eu.kanade.tachiyomi.network.GET
-import eu.kanade.tachiyomi.network.asObservableSuccess
+import eu.kanade.tachiyomi.network.POST
 import eu.kanade.tachiyomi.source.model.FilterList
 import eu.kanade.tachiyomi.source.model.MangasPage
 import eu.kanade.tachiyomi.source.model.Page
 import eu.kanade.tachiyomi.source.model.SChapter
 import eu.kanade.tachiyomi.source.model.SManga
-import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.source.online.HttpSource
 import eu.kanade.tachiyomi.util.asJsoup
-import kotlinx.serialization.json.Json
-import kotlinx.serialization.json.jsonArray
-import kotlinx.serialization.json.jsonObject
-import kotlinx.serialization.json.jsonPrimitive
+import keiyoushi.utils.parseAs
+import okhttp3.CookieJar
 import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MediaType.Companion.toMediaType
 import okhttp3.OkHttpClient
 import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
 import okhttp3.Response
-import org.jsoup.nodes.Document
-import org.jsoup.nodes.Element
-import org.jsoup.parser.Parser
-import rx.Observable
-import uy.kohesive.injekt.injectLazy
-import kotlin.random.Random
-
-class ScanManga : ParsedHttpSource() {
+import java.util.concurrent.TimeUnit
+import java.util.zip.Inflater
 
+class ScanManga : HttpSource() {
     override val name = "Scan-Manga"
 
-    override val baseUrl = "https://www.scan-manga.com"
+    override val baseUrl = "https://m.scan-manga.com"
+    private val baseImageUrl = "https://static.scan-manga.com/img/manga"
 
     override val lang = "fr"
 
     override val supportsLatest = true
 
-    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
-        .addNetworkInterceptor { chain ->
-            val originalCookies = chain.request().header("Cookie") ?: ""
-            val newReq = chain
-                .request()
-                .newBuilder()
-                .header("Cookie", "$originalCookies; _ga=GA1.2.${shuffle("123456789")}.${System.currentTimeMillis() / 1000}")
-                .build()
-            chain.proceed(newReq)
-        }.build()
-
-    private val json: Json by injectLazy()
+    override val client: OkHttpClient = network.client.newBuilder()
```

Comment:
```suggestion
    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
```

This convention is enforced whether the website uses Cloudflare or not.